streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1248,15 +1248,16 @@ protected StreamTask createStreamTask(final TaskId id, final Collection<TopicPar
     }
 
     private Producer<byte[], byte[]> createProducer(final TaskId id) {
-        final Map<String, Object> producerConfigs = config.getProducerConfigs(threadClientId);
```

Comment:
That's right.  Only if client ID is not specified, we set `[clientId] = [applicationId]-[UUID]`.

So you think we need to mention this in upgrade notes, that with EOS turned on, the producer IDs would be `producer.[clientId]-streamThread-[sequence-number]-[taskId]` --

(It think it starts with `producer` and not `thread`? Correct me if I am wrong.)